changelog.lua

```diff
@@ -0,0 +1,374 @@
+  for _, p in ipairs({...}) do
+    path = pl_path.join(path, p)
+  end
+  return path
+end
+
+local function collect_files(folder)
+  local files
+  if pl_path.exists(folder) then
+    files = assert(pl_dir.getfiles(folder))
+    if files then
+      table.sort(files)
+    end
+  end
+  local sorted_files = {}
+  for _, filename in ipairs(files or {}) do
+    if is_yaml(filename) then
+      table.insert(sorted_files, filename)
+    end
+  end
+
+  return sorted_files
+end
+
+
+local function fetch_commit_context(filename)
+  local ctx = {}
+  filename = string.sub(filename, 6)
+  --print(filename)
+  local commit_url = "https://api.github.com/repos/kong/kong/commits?path=%s"
+  local response, status_code, _, status_line = http.request(string.format(commit_url, filename))
+  if status_code ~= 200 then
+    return nil, "failed to fetch commits: " .. status_code .. " " .. status_line
+  end
+
+  --print(response)
+  local res = cjson.decode(response)
+  local commit = res[#res]
+  ctx.commit = {
+    sha = commit.sha,
+    message = commit.commit.message,
+  }
+
+  local pull_url = "https://api.github.com/repos/kong/kong/commits/%s/pulls"
+  local response, status_code, _, status_line = http.request(string.format(pull_url, commit.sha))
+  if status_code ~= 200 then
+    return nil, "failed to fetch pulls: " .. status_code .. " " .. status_line
+  end
+  local res = cjson.decode(response)
+  local pull = res[#res]
+  ctx.pull = {
+    number = pull.number,
+    title = pull.title,
+    body = pull.body,
+  }
+
+  return ctx
+end
+
+local function match_pattern(text, pattern, t)
+  for match in string.gmatch(text, pattern) do
+    table.insert(t, match)
+  end
+end
+
+local function collect_sub_folder(system, sub_folder)
+  local data = {
+    features = {},
+    bugfixes = {},
+    breaking_changes = {},
+    dependencies = {},
+    deprecations = {},
+    performances = {},
+  }
+
+  local map = {
+    feature = "features",
+    bugfix = "bugfixes",
+    breaking_change = "breaking_changes",
+    dependency = "dependencies",
+    deprecation = "deprecations",
+    performance = "performances",
+  }
+
+  local files = collect_files(sub_folder)
+  for _, filename in ipairs(files) do
+    local content = assert(pl_file.read(filename))
+    local entry = assert(lyaml.load(content))
+
+    if is_empty_table(entry.prs) or is_empty_table(entry.jiras) then
+      local ctx, err = fetch_commit_context(filename)
+      if not ctx then
+        error("faield to fetch commit context: " .. err)
+      end
+      --print(cjson.encode(ctx))
+      if is_empty_table(entry.prs) then
+        entry.prs = { ctx.pull.number }
+      end
+      if is_empty_table(entry.jiras) then
+        local jiras = {}
+
+        local sources = { string.upper(ctx.commit.message), string.upper(ctx.pull.body) }
+        for _, text in ipairs(sources) do
+          match_pattern(text, "FTI%-%d+", jiras)
+          match_pattern(text, "KAG%-%d+", jiras)
```

Comment:
We have other projects. Suggest we do not hard code these prefixes.

Reply:
https://github.com/Kong/kong/pull/11613. If the rule is too broad, then how can we avoid matching Sep-2023 ?